core/src/main/scala/org/apache/spark/sql/store/CodeGeneration.scala

```diff
@@ -260,6 +283,32 @@ object CodeGeneration extends Logging {
         "dialect")).asInstanceOf[GeneratedStatement]
   }
 
+
+  private[this] def compileGeneratedIndexUpdate(table: String,
```

Comment:
+ @sumwale 

There is no hot-loop in your code. Did you quantify the difference between the performance with code gen and without code -gen? From what I think, you would not get any benefit. Code generation adds lot of complexity and is a debugging challenge. We should not add code generation unless we are sure about the benefits.